lib/KTable/index.vue

```diff
@@ -0,0 +1,484 @@
+        getAriaSort,
+        SORT_ORDER_ASC,
+        SORT_ORDER_DESC,
+        DATA_TYPE_OTHERS,
+        getHeaderStyle,
+      };
+    },
+    /* eslint-disable kolibri/vue-no-unused-properties */
+    props: {
+      /**
+       * An array of objects `{ label, dataType, minWidth, width }`representing the headers of the table. The `dataType` can be one of `'string'`, `'numeric'`, `'date'`, or `'others'`. `label` and `dataType` are required. `minWidth` and `width` are optional.
+       */
+      headers: {
+        type: Array,
+        required: true,
+        validator: function(value) {
+          return value.every(
+            header =>
+              ['label', 'dataType'].every(key => key in header) &&
+              ['string', 'numeric', 'date', 'others'].includes(header.dataType)
+          );
+        },
+      },
+      /**
+       * An array of arrays representing the rows of the table. Each row should have the same number of elements as the headers array.
+       */
+      rows: {
+        type: Array,
+        required: true,
+      },
+      /**
+       * The caption of the table
+       */
+      caption: {
+        type: String,
+        required: true,
+      },
+      /**
+       * Enables or disables local sorting of the table data.
+       */
+      useLocalSorting: {
+        type: Boolean,
+        default: false,
+      },
+      /**
+       * Enables or disables sorting functionality for the table headers.
+       */
+      sortable: {
+        type: Boolean,
+        default: true,
+      },
+    },
+    data() {
+      return {
+        focusedRowIndex: null,
+        focusedColIndex: null,
+        hoveredRowIndex: null,
+      };
+    },
+    computed: {
+      coreOutlineFocus() {
+        return {
+          ':focus': {
+            ...this.$coreOutline,
+            outlineOffset: '-2px',
+          },
+        };
+      },
+    },
+    methods: {
+      /**
+       * Takes care of
+       *  - keyboard navigation focus trap
+       *  - the arrow keys navigation flow
+       *  - the tab keys navigation flow
+       *  - access to focusable elements within a cell via tab and shift tab keys
+       *  - triggering sort on the enter key
+       *  - header highlight
+       */
+      handleKeydown(event, rowIndex, colIndex) {
+        const key = event.key;
+        const totalRows = this.rows.length;
+        const totalCols = this.headers.length;
+
+        let nextRowIndex = rowIndex;
+        let nextColIndex = colIndex;
+        // Identify all focusable elements inside the current cell
+        const currentCell = this.getCell(rowIndex, colIndex);
+        const focusableElements = currentCell
+          ? Array.from(
+              currentCell.querySelectorAll(
+                'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])'
+              )
+            )
+          : [];
+        const focusedElementIndex = focusableElements.indexOf(document.activeElement);
```

Comment:
We use `currentCell`, `focusableElemens` and `focusedElementIndex` just when we navigate through tabs, so its better to move this into the tabs handler block to avoid these calculations for the rest of keypresses :)